### ISSUE TITLE
DISCUSSION: topic boxes on profile pages via automatic subscriptions upon signup?

### ISSUE THREAD

**Issue by ebarry** [publiclab/plots2#6220]
Hello!!! 🌳🌳🌳🌳🌳🌳🌳🌳

I would like to open a discussion with an observation that i had after reviewing dozens of new profiles pages registered on PublicLab.org as part of my activities to moderate first-time-poster content. This is an adjacent issue to https://github.com/publiclab/plots2/issues/6219 about the duplicate topic boxes. 

On the profile page of someone who has contributed only once, i would not expect to see any topic boxes yet. 🌳 However, i believe that i am seeing new profile pages that already have many, many topic boxes of a variety of our biggest topics -- balloon-mapping, spectrometry, newsletter. (see https://github.com/publiclab/plots2/issues/6219)

I would like to use this issue to discuss the assumptions made about the "automatic subscriptions upon signup" that i think -- although i'm not sure -- are generating these topic boxes that appear on profile pages. 😮 

Something to consider: If i were a new member myself 👩‍🎨, it is quite possible that i would become irritated that spectrometry was showing up all over my profile page for the following reasons 

1. this is not my interest
2. i don't want to be represented this way publicly
3. i specifically don't want to be subscribed to spectrometry 

Specifically regarding the newsletter tag, i have a couple @jywarren questions :) -- 

Automatically subscribing people to the /newsletter tag could be useful if we were in fact posting our monthly newsletter as research notes to the website. Are we? I don't think we are but i'd like to discover that we are!

Or perhaps an automatic subscription to the newsletter tag adds people to our newsletter recipient list in Campaign Monitor? I do not think this is happening, but i'd be excited to discover that it was!

I am interested in hearing your thoughts on the above. thanks!

**Comment by jywarren** [comment]
Hi, actually the auto-subscriptions haven't been implemented yet, and what
you are seeing on new profiles is the result of some very old code, that
fills in the following tags if you have no tags on any posts (and/or no
posts):

    tagnames += ['balloon-mapping', 'spectrometer', 'near-infrared-camera',
'thermal-photography', 'newsletter'] if tagnames.empty? && defaults

https://github.com/publiclab/plots2/blob/d4d69ab4144d1457749ff215c53b63e167cc7142/app/models/user.rb#L157

So this is an unintended consequence.

The later-on plan of auto-subscribing people was roughly intended to
include 5 or so "top topics" and we had assumed air-quality, water-quality,
and a few other top ones from our featured topics (list now in the footer).
Def. we want to get input on which should be auto-subscribed, but we
haven't implemented that yet.

**Comment by jywarren** [comment]
Can we think of any reason we'd want to substitute other "backup" tags for
any purpose? If not, we can simply remove that line and this should resolve
itself. Thanks!

**Comment by ebarry** [comment]
If the backup tags do not equal a subscription, then i can't think of any reason right now. 
I suppose without them, profile pages might look blank, but i think that's preferable. We could just have a message saying "please scroll down to the footer to see some topics you could subscribe to" or something like that

**Comment by jywarren** [comment]
Well, wait, the profile is not showing subscribed-to topics, but topics
within which the author has posted. So if the author hasn't posted anything
that has been tagged, we could show text like:

This person has not posted under any _topics_ (linking to /tags).

We should consider a way to view any untagged posts they may have. I'm not
sure about this, but we could consider something like a /notes/AUTHORNAME
route that simply shows a list.

**Comment by ebarry** [comment]
So perhaps the proposal here is to 
- [ ] remove the very old code, that fills in the following tags if you have no tags on any posts (and/or no posts): tagnames += ['balloon-mapping', 'spectrometer', 'near-infrared-camera',
'thermal-photography', 'newsletter'] if tagnames.empty? && defaults https://github.com/publiclab/plots2/blob/d4d69ab4144d1457749ff215c53b63e167cc7142/app/models/user.rb#L157
- [ ] For those who have not posted any tagged content, consider adding something like a /notes/AUTHORNAME route that simply shows a list on their content on their profile. 

?

**Comment by jywarren** [comment]
Yes, this sounds right!

**Comment by jywarren** [comment]
And linking the `/notes/authorname` route to the new issue opened in #6253 !

**Comment by jywarren** [comment]
OK, closing in favor of https://github.com/publiclab/plots2/pull/6254 and #6253. Thank you!
